challenge_3/c/scottsquatch/src/main.c

```diff
@@ -0,0 +1,94 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+/** Constants **/
+#define MODE_READ  "r"
+#define NUM_ARGS 3
+#define FILE_NAME_ARG 1
+#define ARRAY_SIZE_ARG 2
+
+/** Function Prototypes **/
+int int_compare(const void*, const void*);
+void print_array(int*, int);
+int get_majoriy_element(int*, int);
+int* read_array(char*,int);
+
+int main(int argc, char* argv[])
+{
+  // Get parameters from input
+  if (argc != NUM_ARGS)
+  {
+    // Incorrect number of parameters
+    printf("Incorrect Number of parameters, expected %d but got %d\n",
+      NUM_ARGS - 1, argc - 1);
+    printf("Usage: ./majority_element <array file> <size of array in file>\n");
+  }
+  else
+  {
+    // Read array from file
+    char* fileName = argv[FILE_NAME_ARG];
+    int arraySize = atoi(argv[ARRAY_SIZE_ARG]);
+    int* input_array = read_array(fileName, arraySize);
+
+    // Output to Screen for informational purpose
+    printf("array: ");
+    print_array(input_array, arraySize);
+    printf("\n");
+
+    // Print Majority Element
+    printf("Majority Element: %d\n", get_majoriy_element(input_array, arraySize));
+
+    // Free memory
+    free(input_array);
```

Comment:
Just a tip, I would get in the habit of setting the freed pointer to NULL. Sure, it doesn't mean much as it stands here, but if you ever add code below this and start using that freed pointer, at least you'll crash right away and catch your bug instead of missing a potential security vuln.